app/controllers/api_controller.rb

```diff
@@ -223,7 +223,25 @@ def validate_include_params
   end
 
   def included_relationships
-    IncludeParamHandler.new(params).call
+    @included_relationships ||= IncludeParamHandler.new(params).call
+  end
+
+  def included_db_relationships
+    @included_db_relationships ||= included_relationships.map { |value| cast_relationship_to_symbol_hash(value) } if included_relationships.present?
+  end
+
+  def cast_relationship_to_symbol_hash(value)
+    parts = value.split('.', 2)
+    if parts.length == 1
+      # special hack to allow for the flags table renamed in the backend but not in the FE includes
+      if parts.first == 'flags'
+        :framework_flags
+      else
+        parts.first.to_sym
+      end
+    else
+      { parts.first.to_sym => cast_relationship_to_symbol_hash(parts.last) }
```

Comment:
Ruby doesn't optimise for recursion 😢  so I'd avoid it and prefer mutating state/iterating

Reply:
[shakes fist at ruby]

Reply:
I actually think this process is complex enough to need unit testing in a utility class.

Reply:
Presumably these hashes are unlikely to be more than 3 layers deep so not sure the recursion thing is a big issue. But I agree this does need strong coverage.

Reply:
Just good to not encourage it, yeah